### PR TITLE
PRL-5117-Database call before HWF

### PR DESCRIPTION
**Pull request by sealDebolina**
### JIRA link (if applicable) ##
https://tools.hmcts.net/jira/browse/PRL-5117
### Change description ###
CUI -Backend call to save the data entered until the Help with fees screen
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/steps/c100-rebuild/help-with-fees/need-help-with-fees/routeGuard.ts [hunk 1]
```diff
@@ -2,10 +2,18 @@ import { NextFunction, Response } from 'express';
 
 import { AppRequest } from '../../../../app/controller/AppRequest';
 import { getFeesForC100ApplicationSubmission } from '../../../../app/fees/fees-lookup-api';
+import { C100_CASE_EVENT } from 'app/case/definition';
 
 export const routeGuard = {
   // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
   get: async (req: AppRequest, res: Response, next: NextFunction) => {
+      await req.locals.C100Api.updateCase(
```
Comment (vivek-sekhar) [inline src/main/steps/c100-rebuild/help-with-fees/need-help-with-fees/routeGuard.ts:10]: Handle failure scenario pls

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/c100-rebuild/help-with-fees/need-help-with-fees/routeGuard.ts [hunk 2]
```diff
@@ -2,16 +2,31 @@ import { NextFunction, Response } from 'express';
...
     req.session.userCase = {
       ...(req.session.userCase ?? {}),
       c100ApplicationFees,
     };
     req.session.save(next);
-  },
+  }
+catch(error){
+  req.locals.logger.error('error in update case', error);
+  req.session.save(()=>res.redirect(HOME_URL));
```
Comment (vivek-sekhar) [inline src/main/steps/c100-rebuild/help-with-fees/need-help-with-fees/routeGuard.ts:29]: I think you can redirect the user to same page with some error? can you check with Subha if any error message needs to be shown on the page?

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/c100-rebuild/help-with-fees/need-help-with-fees/routeGuard.ts (hunk 2 above)
Comment (sealDebolina) [reply src/main/steps/c100-rebuild/help-with-fees/need-help-with-fees/routeGuard.ts:29]: not required for this ticket as per discussion with Subha

Comment (sealDebolina) [reply src/main/steps/c100-rebuild/help-with-fees/need-help-with-fees/routeGuard.ts]: error scenario is not required as per discussion with Subha

**Review by AlokDatta — APPROVED** [review]

**Review by vivek-sekhar — APPROVED** [review]

**Pull request merged**